src/mergeDeep.test-d.ts

```diff
@@ -1,49 +1,86 @@
-  const a = { foo: "bar" };
-  const b = { bar: "baz" };
-  const result = mergeDeep(a, b);
+  const result = mergeDeep(
+    { foo: "bar" } as { foo: string },
+    { bar: "baz" } as { bar: string },
+  );
   expectTypeOf(result).toEqualTypeOf<{ foo: string; bar: string }>();
 });
 
 it("merges fully overlapping types", () => {
-  const a = { foo: "bar" };
-  const b = { foo: "baz" };
-  const result = mergeDeep(a, b);
+  const result = mergeDeep(
+    { foo: "bar" } as { foo: string },
+    { foo: "baz" } as { foo: string },
+  );
   expectTypeOf(result).toEqualTypeOf<{ foo: string }>();
 });
 
 it("merges semi-overlapping types", () => {
-  const a = { foo: "bar", x: 1 };
-  const b = { foo: "baz", y: 2 };
-  const result = mergeDeep(a, b);
+  const result = mergeDeep(
+    { foo: "bar", x: 1 } as { foo: string; x: number },
+    { foo: "baz", y: 2 } as { foo: string; y: number },
+  );
   expectTypeOf(result).toEqualTypeOf<{ foo: string; x: number; y: number }>();
 });
 
 it("deeply merges", () => {
-  const a = { foo: { bar: "baz" } };
-  const b = { foo: { qux: "quux" } };
-  const result = mergeDeep(a, b);
+  const result = mergeDeep(
+    { foo: { bar: "bar" } } as { foo: { bar: string } },
+    { foo: { qux: "qux" } } as { foo: { qux: string } },
+  );
   expectTypeOf(result).toEqualTypeOf<{ foo: { bar: string; qux: string } }>();
 });
 
 it("overrides types", () => {
-  const a = { foo: { bar: "baz" } };
-  const b = { foo: "qux" };
-  expectTypeOf(mergeDeep(a, b)).toEqualTypeOf<typeof b>();
-  expectTypeOf(mergeDeep(b, a)).toEqualTypeOf<typeof a>();
+  const a = { foo: { bar: "baz" } } as { foo: { bar: string } };
+  const b = { foo: "qux" } as { foo: string };
+
+  const resultAB = mergeDeep(a, b);
+  expectTypeOf(resultAB).toEqualTypeOf<{ foo: string }>();
+
+  const resultBA = mergeDeep(b, a);
+  expectTypeOf(resultBA).toEqualTypeOf<{ foo: { bar: string } }>();
 });
 
 it("doesn't spread arrays", () => {
-  const a = { foo: ["bar"] as const };
-  const b = { foo: ["baz"] as const };
-  const result = mergeDeep(a, b);
-  expectTypeOf(result).toEqualTypeOf<{ foo: readonly ["baz"] }>();
+  const result = mergeDeep(
+    { foo: ["bar"] } as const,
+    { foo: ["baz"] } as const,
+  );
+  expectTypeOf(result).toEqualTypeOf<{ readonly foo: readonly ["baz"] }>();
 });
 
 it("doesn't recurse into arrays", () => {
-  const a = { foo: [{ bar: "baz" }] };
-  const b = { foo: [{ bar: "hello, world" }] };
-  expectTypeOf(mergeDeep(a, b)).toEqualTypeOf<typeof b>();
+  const result = mergeDeep(
+    { foo: [{ bar: "baz", x: 123 }] } as {
+      foo: Array<{ bar: string; x: number }>;
+    },
+    { foo: [{ bar: "hello, world", y: 456 }] } as {
+      foo: Array<{ bar: string; y: number }>;
+    },
+  );
+  expectTypeOf(result).toEqualTypeOf<{
+    foo: Array<{ bar: string; y: number }>;
+  }>();
+});
+
+it("works with interfaces", () => {
+  // eslint-disable-next-line @typescript-eslint/consistent-type-definitions -- Intentional
```

Comment:
**nit:** we can probably lift this to the file level :shrug:

Reply:
I usually do that once the lint suppression is relevant to more than one test. As long as its isolated I prefer keeping it local.